tests/e2e/system_test/test_cluster_full_and_recovery.py

```diff
@@ -0,0 +1,348 @@
+        self.benchmark_obj.cleanup()
+        self.banchmark_operator_teardown = False
+
+        log.info("Change Ceph full_ratio from from 85% to 95%")
+        change_ceph_backfillfull_ratio(95)
+
+        log.info("Verify PVC2 [CEPH-FS + CEPH-RBD]  are moved to Bound state")
+        helpers.wait_for_resource_state(
+            resource=pvc_obj_blk2, state=constants.STATUS_BOUND, timeout=600
+        )
+        helpers.wait_for_resource_state(
+            resource=pvc_obj_fs2, state=constants.STATUS_BOUND, timeout=600
+        )
+
+        log.info("Verify snapshots moved from false state to true state")
+        snap_fs1_obj.ocp.wait_for_resource(
+            condition="true",
+            resource_name=snap_blk1_obj.name,
+            column=constants.STATUS_READYTOUSE,
+            timeout=300,
+        )
+        snap_blk1_obj.ocp.wait_for_resource(
+            condition="true",
+            resource_name=snap_blk1_obj.name,
+            column=constants.STATUS_READYTOUSE,
+            timeout=300,
+        )
+
+        log.info(f"Creating a PVC from snapshot [restore] {snap_blk1_obj.name}")
+        restore_pvc_blk1_obj = snapshot_restore_factory(
+            snapshot_obj=snap_blk1_obj,
+            size="2Gi",
+            volume_mode=snap_blk1_obj.parent_volume_mode,
+            access_mode=snap_blk1_obj.parent_access_mode,
+            status="",
+        )
+        pod_restore_blk1_obj = pod_factory(
+            interface=constants.CEPHFILESYSTEM,
+            pvc=restore_pvc_blk1_obj,
+            status=constants.STATUS_RUNNING,
+        )
+        pod_restore_blk1_obj.md5 = cal_md5sum(
+            pod_obj=pod_restore_blk1_obj,
+            file_name="fio-rand-write",
+            block=False,
+        )
+
+        log.info(f"Creating a PVC from snapshot [restore] {snap_fs1_obj.name}")
+        restore_pvc_fs1_obj = snapshot_restore_factory(
+            snapshot_obj=snap_fs1_obj,
+            size="2Gi",
+            volume_mode=snap_fs1_obj.parent_volume_mode,
+            access_mode=snap_fs1_obj.parent_access_mode,
+            status="",
+        )
+        pod_restore_fs1_obj = pod_factory(
+            interface=constants.CEPHFILESYSTEM,
+            pvc=restore_pvc_fs1_obj,
+            status=constants.STATUS_RUNNING,
+        )
+        pod_restore_fs1_obj.md5 = cal_md5sum(
+            pod_obj=pod_restore_fs1_obj,
+            file_name="fio-rand-write",
+            block=False,
+        )
+
+        assert pod_restore_fs1_obj.md5 == pod_fs1_obj.md5, (
+            f"md5sum of restore_fs1 {pod_restore_fs1_obj.md5} is not equal "
+            f"to pod_fs1_obj {pod_fs1_obj.md5}"
+        )
+        assert pod_restore_blk1_obj.md5 == pod_blk1_obj.md5, (
+            f"md5sum of restore_blk1 {pod_restore_blk1_obj.md5} is not equal "
+            f"to pod_blk1_obj {pod_blk1_obj.md5}"
+        )
+
+    def verify_used_capacity_greater_than_expected(self, expected_used_capacity, pvc_factory, pod_factory):
```

Comment:
why do we need pvc and pod factory in the know the cluster used capacity ?

Reply:
I tested this scenario many times and sometimes the used capacity is 84.9% .
So I added WA to create a new FIO pod.